mlir/include/mlir/Dialect/Transform/Interfaces/TransformInterfaces.h

```diff
@@ -1089,6 +1096,9 @@ class ErrorCheckingTrackingListener : public TrackingListener {
 
   /// The number of errors that have been encountered.
   int64_t errorCounter = 0;
+
+  /// Latest message from match failure notification.
+  std::string matchFailureMsg = "";
```

Comment:
Can we keep a `Diagnostic`, potentially wrapped in `std::optional/unique_ptr` here? There's more to it than the string...